src/IIIFPresentation/API/Features/Storage/Requests/PostHierarchicalCollection.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System.Data;
+using API.Auth;
+using API.Converters;
+using API.Features.Storage.Helpers;
+using API.Helpers;
+using API.Infrastructure.Requests;
+using API.Settings;
+using AWS.S3;
+using AWS.S3.Models;
+using Core;
+using IIIF.Presentation.V3;
+using IIIF.Presentation.V3.Content;
+using IIIF.Serialisation;
+using MediatR;
+using Microsoft.Extensions.Options;
+using Models.API.General;
+using Repository;
+using Repository.Helpers;
+using DatabaseCollection = Models.Database.Collections;
+using IIdGenerator = API.Infrastructure.IdGenerator.IIdGenerator;
+
+namespace API.Features.Storage.Requests;
+
+public class PostHierarchicalCollection(
+    int customerId,
+    string slug, 
+    UrlRoots urlRoots,
+    string rawRequestBody) : IRequest<ModifyEntityResult<Collection, ModifyCollectionType>>
+{
+    public int CustomerId { get; } = customerId;
+
+    public string Slug { get; } = slug;
+    
+    public UrlRoots UrlRoots { get; } = urlRoots;
+    
+    public string RawRequestBody { get; } = rawRequestBody;
+}
+
+public class PostHierarchicalCollectionHandler(
+    PresentationContext dbContext,    
+    ILogger<CreateCollection> logger,
+    IBucketWriter bucketWriter,
+    IIdGenerator idGenerator,
+    IOptions<ApiSettings> options)
+    : IRequestHandler<PostHierarchicalCollection, ModifyEntityResult<Collection, ModifyCollectionType>>
+{
+    private readonly ApiSettings settings = options.Value;
+    
+    public async Task<ModifyEntityResult<Collection, ModifyCollectionType>> Handle(PostHierarchicalCollection request,
+        CancellationToken cancellationToken)
+    {
+        var splitSlug = request.Slug.Split('/');
+
+        var parentSlug = String.Join(String.Empty, splitSlug.Take(..^1));
```

Comment:
Why use `String` here, rather than `string` alias? Don't think anywhere else in codebase does this?